### ISSUE TITLE
clangd Fails to Resolve Call Hierarchy for Constructors Wrapped by `std::make_unique`

### ISSUE THREAD

**Issue by linzj** [clangd/clangd#2242]
**Description:**

I'm using the clangd plugin in VSCode with clangd version 18. I've encountered an issue where calls to constructors wrapped in `std::make_unique` do not appear in the call hierarchy.

### Steps to Reproduce:
1. Create a class (e.g., `Foo`).
2. Construct objects using `std::make_unique<Foo>(...)`.
3. Attempt to view the call hierarchy for the constructor.

### Expected Behavior:
The call hierarchy should display all calls to the constructor, including those wrapped by `std::make_unique`.

### Actual Behavior:
The call hierarchy does not show the constructor call when wrapped in `std::make_unique`. However, using the `new` keyword (e.g., `new Foo(...)`) works as expected and shows up in the call hierarchy.

### Workaround:
Currently, the only workaround is to use "Find All References" to locate the constructor calls.

### Example Code:
```cpp
class Foo {
    // Class definition
};

void bar() {
    auto foo = std::make_unique<Foo>(/* constructor arguments */);
}
```

### Additional Information:
- clangd version: 18
- VSCode version: 1.95.3
- Operating system: Windows 10

Looking forward to a fix or any guidance on this issue. Thank you!

**Comment by HighCommander4** [comment]
> Currently, the only workaround is to use "Find All References" to locate the constructor calls.

To clarify, I assume you mean using find-references on the **class**, since find-references on the **constructor** also does not find the use of `make_unique`.

This is a known issue tracked in https://github.com/clangd/clangd/issues/716. We can keep this one open to track any call hierarchy specific work that may be needed once #716 is resolved.

**Comment by linzj** [comment]
Yes, your clarification is correct. Thank you for providing this additional information and for pointing out that it's a known issue being tracked in #716. I appreciate your help!

**Comment by HighCommander4** [comment]
Confirmed that this is fixed in clangd 22 by #716.
